framework/src/modules/interoperability/base_interoperability_method.ts

```diff
@@ -92,33 +92,40 @@ export abstract class BaseInteroperabilityMethod<
 		return this.stores.get(TerminatedOutboxStore).get(context, chainID);
 	}
 
+	public async getChannelCommon(context: ImmutableMethodContext, chainID: Buffer) {
+		const mainchainID = getMainchainID(chainID);
+		const ownChainAccount = await this.getOwnChainAccount(context);
+		const hasChainAccount = await this.stores.get(ChainAccountStore).has(context, chainID);
+
+		let updatedChainID = chainID;
+		if (!ownChainAccount.chainID.equals(mainchainID) && !hasChainAccount) {
```

Comment:
maybe add comments here from the LIP `# Processing on a sidechain` and `# Check for direct channel.`